pyccel/ast/operators.py

```diff
@@ -945,14 +936,21 @@ def _calculate_dtype(*args):
         -------
         dtype : DataType
             The underlying datatype of the object.
-        precision : int
-            The precision of the datatype of the object.
         class_type : DataType
             The Python type of the object.
         """
-        dtype = NativeBool()
-        precision = -1
-        return dtype, precision, dtype
+        dtype = PythonNativeBool()
+        possible_class_types = set(a.class_type for a in args \
+                        if isinstance(a.class_type, ContainerType))
+        if len(possible_class_types) == 0:
+            class_type = dtype
+        elif len(possible_class_types) == 1:
+            cls = type(possible_class_types.pop())
+            class_type = cls(dtype)
+        else:
+            raise NotImplementedError("Can't deduce type for comparison operator"
+                                      " with multiple containers {', '.join(args)}")
+        return class_type
```

Comment:
According to the docstring this function should return `(dtype, class_type)`.

If `class_type` is a tuple we are in a funny situation...

Reply:
This is a mistake. This was the case mid-way through my changes. I'm surprised that NumPyDoc doesn't flag it though. Fixed : ad383a79f